elementary/monitor/data_monitoring/data_monitoring.py

```diff
@@ -36,7 +39,12 @@ def __init__(
         latest_invocation = self.get_latest_invocation()
         self.project_name = latest_invocation.get("project_name")
         dbt_pkg_version = latest_invocation.get("elementary_version")
+        self.warehouse_info = self._get_warehouse_info(
+            hash_id=isinstance(tracking, AnonymousTracking)
+        )
         if tracking:
+            if self.warehouse_info:
+                tracking.register_warehouse_group(self.warehouse_info)
```

Comment:
You can just use `_register_group` here instead of having a specific method for that that is only used once